dsgrid/dataset/dataset_schema_handler_standard.py

```diff
@@ -73,11 +79,26 @@ def _check_lookup_data_consistency(self, config: DatasetConfig, load_data_lookup
                 raise DSGInvalidDataset(
                     f"load_data_lookup records do not match dimension records for {name}"
                 )
+            dim_records_list.append(dim_records)
+            dim_names.append(name)
+
+        dim_records_combo = set(itertools.product(*dim_records_list))
+        data_records_combo = get_unique_values(load_data_lookup, dim_names)
+        if dim_records_combo != data_records_combo:
+            missing = dim_records_combo.difference(data_records_combo)
+            logger.error(
+                "load_data_lookup is missing %s dimension combination(s): \n%s",
+                len(missing),
+                missing,
+            )
+            raise DSGInvalidDataset(
+                f"load_data_lookup records do not match dimension records for dimension combinations = {dim_names}"
+            )
```

Comment:
This code is probably also subject to Dan's questions about whether we should be doing this kind of checking in Spark